refactor(RSSHandlerRoutine): switch to Instant

christolis opened the pull request:
From a recent [conversation](https://discord.com/channels/272761734820003841/895717328359153664/1455140764903538881):

```log
[13:09]Zabuzard: while at it, ZonedDateTime should be replaced by Instant in that file
[13:10]Zabuzard: does someone want to tackle that quickly or else we need to make a github issue 🙂
[13:56]christolis: what makes this change worthwhile?
[14:01]Zabuzard: its just the wrong class for the job
[14:01]Zabuzard: there is no reason to use ZDT in this case
[14:01]Zabuzard: Instant is the correct class
[14:02]Zabuzard: shouldnt require any deeper changes in this case, just a quick replacement should work```

christolis committed:
refactor(RSSHandlerRoutine): switch to Instant

Signed-off-by: Chris Sdogkos <work@chris-sdogkos.com>

christolis committed:
Pull in hotfix so that CI checks pass

christolis committed:
Remove unnecessary DateTimeFormatter variable

Signed-off-by: Chris Sdogkos <work@chris-sdogkos.com>

tj-wazei reviewed (dismissed):
Looks reasonable to me.

ankitsmt211 reviewed (changes requested):
everything looks good but we should handle the outdated timestamps stored in DB

christolis committed:
Address code review comments from @Zabuzard

Signed-off-by: Chris Sdogkos <work@chris-sdogkos.com>

christolis committed:
Handle updating of wrongly formatted timestamps

Handle rewriting timestamps from the specific RSS feed's format to a
more unified Instant string format.

If a DateTimeParseException is thrown while attempting to work with the
date format, use that opportunity to attempt to parse that original date
with the given date format pattern, then convert it into a string using
'Instant#toString'.

If the conversion fails, simply return an empty Optional and let the
rest of the code handle it from there since the new value will be
overwritten in any case.

Signed-off-by: Chris Sdogkos <work@chris-sdogkos.com>

ankitsmt211 reviewed (changes requested):
It seems my assumptions were wrong, time parsing wouldn't blow up
I have looked at the database, rss_feed stores Instant friendly timestamps for our active rss_feeds using "yyyy-MM-dd'T'HH:mm:ssXXX" datetime format from configuration.

So I have pointed out the error handling that we wouldn't need anymore. Apologies for the confusion, should have looked into this more carefully before putting out any comments.
